Source/AwsCommonRuntimeKit/mqtt/Mqtt5Types.swift

```diff
@@ -653,9 +650,9 @@ public class LifecycleDisconnectData {
     public let crtError: CRTError
 
     /// Data model of an `MQTT5 DISCONNECT <https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901205>`_ packet.
-    public let disconnectPacket: DisconnectPacket
+    public let disconnectPacket: DisconnectPacket?
 
-    public init (crtError: CRTError, disconnectPacket: DisconnectPacket) {
+    public init (crtError: CRTError, disconnectPacket: DisconnectPacket? = nil) {
```

Comment:
The same questions as above: Could DisconnectPacket be nil?

Reply:
DisconnectPacket is only available in scenario where the IoT broker sends a disconnect packet. Other disconnections (e.g. user is shutting down client while connected) results in a disconnect lifecycle event without a disconnect packet.